python/graphstorm/eval/evaluator.py

```diff
@@ -1236,6 +1240,16 @@ def _get_major_score(self, score):
             major_score = score[self.major_etype]
         return major_score
 
+    def _get_early_stop_score(self, val_score: dict[dict]) -> float:
+        # First metric is the key eval metric
+        metric = self.metric_list[0]
+        assert metric in val_score, \
+            f"{metric} is used to guide the early stop. " \
+            f"But the evaluation scores do not report {metric}." \
+            f"Instead we get {val_score}"
+        # call _get_major_score to get the score for early stop.
+        return self._get_major_score(val_score[metric])
```

Comment:
Should be `return val_score[metric]` ?

Reply:
For per edge type evaluator, we call `_get_major_score`. By default `_get_major_score` will return the average score of all the edge types. But if a user provides the `major_etype`, then the score of `major_etype` will be returned.